### DISCUSSION TITLE
A repository rewrite involves reappend all backtrack links to rewrited commits with issue links

### DISCUSSION THREAD

**Discussion by andry81** [community/community#49227] [Repositories]
### Select Topic Area

Bug

### Body

I am not sure is that a bug, but nevertheless it brings spam into the Issues with backtrack links to commits being rewrited.

Steps to reproduce:
1. Create a repo.
2. Add commits containing existing issue links (`https://github.com/<user>/<repo>/issues/<number>`) in the source files (readme, changelog, whatever).
3. After the push the respective issues will contain the backtrack links to the commits containing issue links.
4. Rewrite the repo fixing for example user email or name.
5. The backtrack links in the issues added again.

I don't know could that be currently reproducible, but I was noticed this a quite time ago.

**Comment by airtower-luna** [comment]
That behavior makes sense: When you rewrite repository history, you create new commits (as you can see by the changed commit hash) that are very similar in content to the old ones. Those new commits link to the issues again.

**Reply by andry81** [reply]
Why not cleanup the old one links if related commits is purged from the repo?

**Reply by airtower-luna** [reply]
Have those commits actually been purged? By default they aren't, you'll need to ask GitHub support to do that, see the red warning box here: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

**Reply by andry81** [reply]
Is there any API to purge them on GitHub? I think the trash commits would consume the storage space on GitHub and GitHub really don't want to store them too long. Would be better to have some interface to enforce purge it to avoid storage limit hit.

**Reply by airtower-luna** [reply]
No, the documentation I linked explicitly says to contact support.

**Comment by andry81** [comment]
@airtower-luna Those are not new commits, but rewritten commits. And this is a bad design, because any repo rewrite would spam in a random repo issues discussion referenced in a source file of the being rewritted repo.
